clc/modules/object-storage/src/main/java/com/eucalyptus/objectstorage/ObjectStorageGateway.java

```diff
@@ -2013,113 +2055,317 @@ public DeleteBucketTaggingResponseType deleteBucketTagging(DeleteBucketTaggingTy
 
   @Override
   public GetBucketCorsResponseType getBucketCors(GetBucketCorsType request) throws S3Exception {
-    Bucket bucket = getBucketAndCheckAuthorization(request);
-
-    // Get the CORS configuration from the back-end and copy results in.
-    GetBucketCorsResponseType reply = (GetBucketCorsResponseType) request.getReply();
-
     Boolean isEmpty = false;
+    String bucketName = "nullBucket";
+    GetBucketCorsResponseType response = null;
 
+    // Let most NullPointerExceptions be handled by the catch block.
     try {
+      Bucket bucket = getBucketAndCheckAuthorization(request);
+      bucketName = request.getBucket();
+      response = (GetBucketCorsResponseType) request.getReply();
+
       CorsConfiguration corsConfiguration = new CorsConfiguration();
       List<CorsRule> responseRules = BucketCorsManagers.getInstance().getCorsRules(bucket.getBucketUuid());
       isEmpty = responseRules.isEmpty();
```

Comment:
Null checks would be better than letting the try-catch block handle an NPE. Encapsulating the logic in the try-catch block makes it more readable imho. 

```
  @Override
  public GetBucketCorsResponseType getBucketCors(GetBucketCorsType request) throws S3Exception {
    GetBucketCorsResponseType response = (GetBucketCorsResponseType) request.getReply();
    String bucketName = request.getBucket();

    // Let most NullPointerExceptions be handled by the catch block.
    try {
      Bucket bucket = getBucketAndCheckAuthorization(request);
      CorsConfiguration corsConfiguration = new CorsConfiguration();
      List<CorsRule> responseRules = BucketCorsManagers.getInstance().getCorsRules(bucket.getBucketUuid());
      if (responseRules != null && !responseRules.isEmpty()) {
        corsConfiguration.setRules(responseRules);
        response.setCorsConfiguration(corsConfiguration);
      } else {
        throw new NoSuchCorsConfigurationException(bucket.getBucketName());;    
      }
    } catch (S3Exception s3e) {
      LOG.warn("Caught S3Exception while getting the CORS configuration for bucket <" + 
          bucketName + ">, CorrelationId: " + Contexts.lookup().getCorrelationId() + 
          ", responding to client with: ", s3e);
      throw s3e;
    } catch (Exception ex) {
      LOG.warn("Caught general exception while getting the CORS configuration for bucket <" + 
          bucketName + ">, CorrelationId: " + Contexts.lookup().getCorrelationId() + 
          ", responding to client with 500 InternalError because of: ", ex);
      throw new InternalErrorException(bucketName, ex);
    }

    return response;
  }
```